rust/main/hyperlane-base/src/contract_sync/mod.rs

```diff
@@ -179,7 +181,12 @@ where
                     }
                 };
 
-                let logs = self.dedupe_and_store_logs(logs, stored_logs_metric).await;
+                let logs_result = self.dedupe_and_store_logs(logs, stored_logs_metric).await;
+                let (logs, error) = match logs_result {
+                    Ok(logs) => (logs, false),
+                    Err(logs) => (logs, true),
+                };
```

Comment:
this usage pattern also indicates a `bool` tuple item would be cleaner than a `Result`

Reply:
Fixed